src/components/ui/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+*
+* This software is licensed under the Apache License, Version 2.0 (the
+* "License") as published by the Apache Software Foundation.
+*
+* You may not use this file except in compliance with the License. You may
+* obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+
+import React, { useEffect } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+
+import { apiKeyLocalStorageKey } from "../../../constants";
+import { getValueFromStorage } from "../../../storageService";
+
+const ProtectedRoute = () => {
+  const [shouldRedirect, setShouldRedirect] = React.useState(false);
+
+  useEffect(() => {
+    const apiKey = getValueFromStorage(apiKeyLocalStorageKey);
+    if (apiKey === null || apiKey.length === 0) {
+      setShouldRedirect(true);
+    }
+  }, []);
+
+  if (shouldRedirect) {
+    return <><Navigate to="/" /></>;
+  }
+
+  return <Outlet />;
+};
+
+export default ProtectedRoute;
```

Comment:
Just have an HOC that checks for the API key and renders a child if its found or redirects to "/" if it doesnt. Dont rely this heavily on the router

Reply:
Updated